tutorials/04-hidden-markov-model/index.qmd

```diff
@@ -190,4 +195,85 @@ stationary. We can use the diagnostic functions provided by [MCMCChains](https:/
 heideldiag(MCMCChains.group(chn, :T))[1]
 ```
 
-The p-values on the test suggest that we cannot reject the hypothesis that the observed sequence comes from a stationary distribution, so we can be reasonably confident that our transition matrix has converged to something reasonable.
\ No newline at end of file
+The p-values on the test suggest that we cannot reject the hypothesis that the observed sequence comes from a stationary distribution, so we can be reasonably confident that our transition matrix has converged to something reasonable.
+
+## Efficient Inference With The Forward Algorithm
+
+While the above method works well for the simple example in this tutorial, some users may desire a more efficient method, especially when their model is more complicated. One simple way to improve inference is to marginalize out the hidden states of the model with an appropriate algorithm, calculating only the posterior over the continuous random variables. Not only does this allow more efficient inference via Rao-Blackwellization, but now we can sample our model with `NUTS()` alone, which is usually a much more performant MCMC kernel.
+
+Thankfully, [HiddenMarkovModels.jl](https://github.com/gdalle/HiddenMarkovModels.jl) provides an extremely efficient implementation of many algorithms related to Hidden Markov Models. This allows us to re-write our model as:
+
+```{julia}
+#| output: false
+using HiddenMarkovModels
+using FillArrays
+using LinearAlgebra
+using LogExpFunctions
+
+
+@model function BayesHmm2(y, K)
+    m ~ Bijectors.ordered(MvNormal([1.0, 2.0, 3.0], 0.5I))
+    T ~ filldist(Dirichlet(Fill(1/K, K)), K)
+
+    hmm = HMM(softmax(ones(K)), copy(T'), [Normal(m[i], 0.1) for i in 1:K])
+    Turing.@addlogprob! logdensityof(hmm, y)
+end
+
+chn2 = sample(BayesHmm2(y, 3), NUTS(), 1000)
+```
+
+
+We can compare the chains of these two models, confirming the posterior estimate is similar (modulo label switching concerns with the Gibbs model):
+```{julia}
+#| code-fold: true
+#| code-summary: "Plotting Chains"

```

Comment:
I'll defer to you here, but I really think that the first code block especially makes the page harder to read. In this case it's probably not a big deal, but I don't want to take a strong stance against all code folding because there may be more extreme cases in the future where its useful (e.g. large amounts of plotting code in the middle of a tutorial that disrupts the flow for the reader).